pkg/sfu/buffer/buffer.go

```diff
@@ -328,41 +344,58 @@ func (b *Buffer) calc(pkt []byte, arrivalTime int64) {
 func (b *Buffer) updateStreamState(p *rtp.Packet, pktSize int, arrivalTime int64, isRTX bool) {
 	sn := p.SequenceNumber
 
-	if b.stats.PacketCount == 0 {
-		b.highestSN = sn - 1
+	if !b.started {
+		b.started = true
+		b.highestSN = sn
 
 		b.lastReport = arrivalTime
 
 		b.rrSnapshot = &receiverReportSnapshot{
-			extSeqNum:       uint32(sn) - 1,
-			packetsReceived: 0,
-			packetsLost:     0,
-			lastLossRate:    0.0,
-		}
-	}
-
-	diff := sn - b.highestSN
-	if diff > (1 << 15) {
-		// out-of-order, remove it from nack queue
-		if b.nacker != nil {
-			b.nacker.Remove(sn)
+			extHighestSeqNum: uint32(sn) - 1,
+			packetsLost:      0,
+			lastLossRate:     0.0,
 		}
 	} else {
-		if b.nacker != nil && diff > 1 {
-			for lost := b.highestSN + 1; lost != sn; lost++ {
-				b.nacker.Push(lost)
+		diff := sn - b.highestSN
+		if diff > (1 << 15) {
+			if !isRTX && b.stats.TotalPacketsLost != 0 {
+				b.stats.TotalPacketsLost--
 			}
-		}
 
-		if sn < b.highestSN && b.stats.PacketCount > 0 {
-			b.cycle++
-		}
+			// out-of-order, remove it from nack queue
+			if b.nacker != nil {
+				b.nacker.Remove(sn)
+			}
+		} else {
+			b.stats.TotalPacketsLost += (uint32(diff) - 1)
+			if b.nacker != nil && diff > 1 {
+				for lost := b.highestSN + 1; lost != sn; lost++ {
+					b.nacker.Push(lost)
+				}
+			}
 
-		b.highestSN = sn
+			if sn < b.highestSN {
+				b.cycle++
+			}
+
+			b.highestSN = sn
+		}
 	}
 
-	b.stats.PacketCount++
-	b.stats.TotalBytes += uint64(pktSize)
+	switch {
+	case isRTX:
+		b.stats.TotalRetransmitPackets++
```

Comment:
Maintaining one of these `stats` struct to keep a bunch of stats about an RTP stream (SSRC). So, there is one per `sfu.Buffer` in the up stream  direction. And one in `sfu.DownTrack` for the down stream tracks.